proxy/src/telemetry/metrics/serve.rs

```diff
@@ -63,22 +77,73 @@ impl Service for Serve {
             let mut writer = GzEncoder::new(Vec::<u8>::new(), CompressionOptions::fast());
             write!(&mut writer, "{}", *metrics)
                 .and_then(|_| writer.finish())
-                .map(|body| {
-                    Response::new()
-                        .with_header(ContentEncoding(vec![Encoding::Gzip]))
-                        .with_header(ContentType::plaintext())
-                        .with_body(Body::from(body))
+                .map_err(ServeError::from)
+                .and_then(|body| {
+                    Response::builder()
+                        .header(header::CONTENT_ENCODING, "gzip")
+                        .header(header::CONTENT_TYPE, "text/plain")
+                        .body(Body::from(body))
+                        .map_err(ServeError::from)
```

Comment:
This shouldn't error, as the headers you've included are valid. I'd probably rather this panic loudly, instead of silently aborting the connection and carrying on. For that reason, I'd also suggest that `ServeError` **not** have a `From<http::Error>`.

Reply:
This function doesn't return `ServeError`s, they're all handled at the end of the function:
https://github.com/runconduit/conduit/blob/21ce0747ee0f0708ecf90d503ad243b504c63933/proxy/src/telemetry/metrics/serve.rs#L101-L108

`ServeError` is only used to consolidate the error handling for the compressed and uncompressed branches in one place. However, if you think it would be clearer just to do it inline, I'd be happy to make that change.

Reply:
Oh woops, I missed that. Carry on then XD